### PR TITLE
Fix bug 1308645: Improve frontend security

### PR DESCRIPTION
**Pull request by mathjazz**

### CODE REVIEW

**Comment by marumari** [comment]
To mirror the comment on bmo:


```
Content-Security-Policy: default-src 'none'; child-src https:; connect-src 'self'; font-src 'self'; frame-src https:; img-src 'self' https://*.wp.com/pontoon.mozilla.org/ https://ssl.google-analytics.com https://www.gravatar.com/avatar/; script-src 'self' https://login.persona.org 'sha256-x3niK4UU+vG6EGT2NK2rwi2j/etQodJd840oRpEnqd4=' 'sha256-fDsgbzHC0sNuBdM4W91nXVccgFLwIDkl197QEca/Cl4=' https://ssl.google-analytics.com/ga.js; style-src 'self' 'unsafe-inline'
```

Would let you frame any site available over https:, blocking ones over http:.  Add http: to child-src and frame-src to allow all websites to be framed.

**Comment by mathjazz** [comment]
OK, final update:
* CSP updated to allow all HTTPS-based project websites to be framed.
* Django middleware settings updated so headr actuall contains HSTS and XCTO.
* I also added the `x-xss-protection: 1; mode=block` header.

[Stage](https://mozilla-pontoon-staging.herokuapp.com/) seems to be working fine.

@marumari @jotes r?

**Comment by marumari** [comment]
Looks great to me!  Thanks for your quick work on this!

File: pontoon/settings/base.py [hunk 1]
```diff
@@ -487,9 +486,45 @@ def _allowed_hosts():
...
+
+# Content-Security-Policy headers
+CSP_DEFAULT_SRC = ("'none'",)
+CSP_CHILD_SRC = ("https:",)
+CSP_FRAME_SRC = ("https:",)  # Older browsers
+CSP_CONNECT_SRC = ("'self'",)
+CSP_FONT_SRC = ("'self'",)
+CSP_IMG_SRC = (
+    "'self'",
+    "https://*.wp.com/pontoon.mozilla.org/",
```
Comment (marumari) [inline pontoon/settings/base.py]: It seems to be largely the same as this image, fwiw: https://pontoon.mozilla.org/static/img/anon.0a768246b7e0.jpg

Comment (mathjazz) [inline pontoon/settings/base.py]: Of course, it's the default Gravatar - thanks!

**Comment by marumari** [comment]
Thank you so much for your work on this!  It's hugely appreciated!

**Comment by mathjazz** [comment]
Thank *you* for being super helpful and responsive and providing the solution withing the bug report!

**Pull request merged**